[release/2.13] FakeTensor error message regression — assertRaisesRegex pattern outdated

test_cases:
  - op_ut,third_party.torch-xpu-ops.test.xpu.test_fake_tensor_xpu.FakeTensorTest,test_add_one_dim_single_elem_cpu_with_cuda_tensor
  - op_ut,third_party.torch-xpu-ops.test.xpu.test_fake_tensor_xpu.FakeTensorTest,test_op_with_zero_dim_bypassed
  - op_ut,third_party.torch-xpu-ops.test.xpu.test_fake_tensor_xpu.FakeTensorPreferDeviceType,test_fake_tensor_prefer_device_type

## Root Cause

PyTorch upstream changed the `FakeTensor` device mismatch error message. The XPU test file uses `assertRaisesRegex` with the old message pattern, which no longer matches.

**Old message (expected by XPU test):**
```
"Unhandled FakeTensor Device Propagation for.*"
```

**New message (actual):**
```
"Expected all tensors to be on the same device, but found at least two devices, cpu and xpu:0!"
```

The underlying FakeTensor behavior is correct — it properly detects the device mismatch. Only the error message text changed upstream.

**Fix:** Update the `assertRaisesRegex` patterns in `test_fake_tensor_xpu.py` to match the new error message, or sync the test file with the latest upstream `test_fake_tensor.py`.

## Reproducer

```bash
source ~/.bashrc && cd third_party/torch-xpu-ops/test/xpu && python -m pytest -sxv test_fake_tensor_xpu.py -k "test_add_one_dim_single_elem_cpu_with_cuda_tensor"
```

```bash
source ~/.bashrc && cd third_party/torch-xpu-ops/test/xpu && python -m pytest -sxv test_fake_tensor_xpu.py -k "test_op_with_zero_dim_bypassed"
```

```bash
source ~/.bashrc && cd third_party/torch-xpu-ops/test/xpu && python -m pytest -sxv test_fake_tensor_xpu.py -k "test_fake_tensor_prefer_device_type"
```

## Representative Error

```
AssertionError: "Unhandled FakeTensor Device Propagation for.*" does not match
"Expected all tensors to be on the same device, but found at least two devices, cpu and xpu:0!"
```

--- a/test/xpu/test_fake_tensor_xpu.py
+++ b/test/xpu/test_fake_tensor_xpu.py
@@ -33,6 +33,7 @@ from torch._subclasses.fake_tensor import (
     extract_tensor_metadata,
     FakeTensor,
     FakeTensorConverter,
+    FakeTensorDeviceMismatchError,
     FakeTensorMode,
     MetadataMismatchError,
     unset_fake_temporarily,
@@ -245,7 +246,8 @@ class FakeTensorTest(TestCase):
         fake_y = mode.from_tensor(y)
 
         with self.assertRaisesRegex(
-            RuntimeError, "Expected all tensors to be on the same device"
+            FakeTensorDeviceMismatchError,
+            "Expected all tensors to be on the same device"
         ) as exc:
             torch.nextafter(fake_x, fake_y)
 
@@ -278,7 +280,8 @@ class FakeTensorTest(TestCase):
             y = torch.randn(10, device=device_type)
 
             with self.assertRaisesRegex(
-                RuntimeError, "Expected all tensors to be on the same device"
+                FakeTensorDeviceMismatchError,
+                "Expected all tensors to be on the same device"
             ) as exc:
                 x + y
 
@@ -2676,7 +2679,8 @@ class FakeTensorPreferDeviceType(TestCase):
 
             # Without the config, this should raise a device mismatch error
             with self.assertRaisesRegex(
-                RuntimeError, "Expected all tensors to be on the same device"
+                RuntimeError,
+                "Expected all tensors to be on the same device"
             ):
                 mixed_device_op(cuda_tensor, None)
 
@@ -2714,7 +2718,8 @@ class FakeTensorPreferDeviceType(TestCase):
 
             # After exiting the config context, should raise error again
             with self.assertRaisesRegex(
-                RuntimeError, "Expected all tensors to be on the same device"
+                RuntimeError,
+                "Expected all tensors to be on the same device"
             ):
                 mixed_device_op(cuda_tensor, None)
 

--- a/test/xpu/test_fake_tensor_xpu.py
+++ b/test/xpu/test_fake_tensor_xpu.py
@@ -245,7 +245,7 @@ class FakeTensorTest(TestCase):
         fake_y = mode.from_tensor(y)
 
         with self.assertRaisesRegex(
-            RuntimeError, "Unhandled FakeTensor Device Propagation for.*"
+            RuntimeError, "Expected all tensors to be on the same device"
         ) as exc:
             torch.nextafter(fake_x, fake_y)
 
@@ -278,7 +278,7 @@ class FakeTensorTest(TestCase):
             y = torch.randn(10, device=device_type)
 
             with self.assertRaisesRegex(
-                RuntimeError, "Unhandled FakeTensor Device Propagation for.*"
+                RuntimeError, "Expected all tensors to be on the same device"
             ) as exc:
                 x + y
 
@@ -2676,7 +2676,7 @@ class FakeTensorPreferDeviceType(TestCase):
 
             # Without the config, this should raise a device mismatch error
             with self.assertRaisesRegex(
-                RuntimeError, "Unhandled FakeTensor Device Propagation"
+                RuntimeError, "Expected all tensors to be on the same device"
             ):
                 mixed_device_op(cuda_tensor, None)
 
@@ -2714,7 +2714,7 @@ class FakeTensorPreferDeviceType(TestCase):
 
             # After exiting the config context, should raise error again
             with self.assertRaisesRegex(
-                RuntimeError, "Unhandled FakeTensor Device Propagation"
+                RuntimeError, "Expected all tensors to be on the same device"
             ):
                 mixed_device_op(cuda_tensor, None)
 

--- a/test/xpu/test_fake_tensor_xpu.py
+++ b/test/xpu/test_fake_tensor_xpu.py
@@ -247,7 +247,7 @@ class FakeTensorTest(TestCase):
 
         with self.assertRaisesRegex(
             FakeTensorDeviceMismatchError,
-            "Expected all tensors to be on the same device"
+            "Expected all tensors to be on the same device",
         ) as exc:
             torch.nextafter(fake_x, fake_y)
 
@@ -281,7 +281,7 @@ class FakeTensorTest(TestCase):
 
             with self.assertRaisesRegex(
                 FakeTensorDeviceMismatchError,
-                "Expected all tensors to be on the same device"
+                "Expected all tensors to be on the same device",
             ) as exc:
                 x + y
 
@@ -2679,8 +2679,7 @@ class FakeTensorPreferDeviceType(TestCase):
 
             # Without the config, this should raise a device mismatch error
             with self.assertRaisesRegex(
-                RuntimeError,
-                "Expected all tensors to be on the same device"
+                RuntimeError, "Expected all tensors to be on the same device"
             ):
                 mixed_device_op(cuda_tensor, None)
 
@@ -2718,8 +2717,7 @@ class FakeTensorPreferDeviceType(TestCase):
 
             # After exiting the config context, should raise error again
             with self.assertRaisesRegex(
-                RuntimeError,
-                "Expected all tensors to be on the same device"
+                RuntimeError, "Expected all tensors to be on the same device"
             ):
                 mixed_device_op(cuda_tensor, None)
 

